File: src/main/java/com/mudhut/nudge/users/repositories/PasswordResetTokenRepository.java
package com.mudhut.nudge.users.repositories;

import org.springframework.data.jpa.repository.JpaRepository;

import com.mudhut.nudge.users.entities.PasswordResetToken;
import com.mudhut.nudge.users.entities.User;

import java.util.List;
import java.util.Optional;

public interface PasswordResetTokenRepository extends JpaRepository<PasswordResetToken, Long> {
    Optional<PasswordResetToken> findByToken(String token);

    List<PasswordResetToken> findAllByUserAndUsedFalse(User user);
}

File: src/main/kotlin/com/mudhut/nudge/users/repositories/PasswordResetTokenRepository.kt
package com.mudhut.nudge.users.repositories

import com.mudhut.nudge.users.entities.PasswordResetToken
import com.mudhut.nudge.users.entities.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface PasswordResetTokenRepository : JpaRepository<PasswordResetToken, Long> {
    fun findByToken(token: String): Optional<PasswordResetToken>
    fun findAllByUserAndUsedFalse(user: User): List<PasswordResetToken>
}
